client/src/app/site/motions/components/motion-import-list/motion-import-list.component.ts

```diff
@@ -0,0 +1,265 @@
+    styleUrls: ['./motion-import-list.component.scss']
+})
+export class MotionImportListComponent extends BaseViewComponent {
+    /**
+     * The data source for a table. Requires to be initialised with a BaseViewModel
+     */
+    public dataSource: MatTableDataSource<NewMotionEntry>;
+
+    public hasFile = false;
+    public selectedEncoding: string;
+    public selectedTextSeparator: string;
+
+    public invalidReasons = {
+        title: 'Motion has no title',
+        text: 'Motion has no text',
+        duplicate: 'A motion with this identifier already exists'
+    };
+
+    /**
+     * indicator on which elements to hide
+     */
+    public hidden = {done: true, errors: false };
+
+    /**
+     * The table itself
+     */
+    @ViewChild(MatTable)
+    protected table: MatTable<NewMotionEntry>;
+
+    /**
+     * Returns the amount of total item successfully parsed
+     */
+    public get totalCount(): number {
+        return (this.importer && this.hasFile) ? this.importer.summary.total : null;
+    }
+
+    public get encodings(): string[] {
+        return this.importer.encodings;
+    }
+
+    public get columnSeparators(): {name: string, value: string}[] {
+        return this.importer.columnSeparators;
+    }
+
+    /**
+     * Returns the amount of import items that will be imported
+     */
+    public get newCount(): number {
+        return (this.importer && this.hasFile) ? this.importer.summary.new : null;
+    }
+
+    public get nonImportableCount(): number {
+        if (this.importer && this.hasFile) {
+            return this.importer.summary.errors + this.importer.summary.duplicates;
+        }
+        return null;
+    }
+
+    public get doneCount(): number {
+        return (this.importer && this.hasFile) ? this.importer.summary.done : null;
+    }
+
+    /**
+     * Constructor for list view bases
+     * @param titleService the title serivce
+     * @param translate the translate service
+     * @param matSnackBar
+     * @param importer: The user csv import service
+     */
+    public constructor( titleService: Title, public translate: TranslateService,
+        matSnackBar: MatSnackBar, private importer: MotionImportService,
+        private motionCSVExport: MotionCsvExportService) {
+        super(titleService, translate, matSnackBar);
+        this.initTable();
+        this.selectedEncoding = this.encodings[0];
+        this.importer.errorEvent.subscribe( message => this.raiseError(message) );
+    }
+
+    /**
+     * Children need to call this in their init-function.
+     * Calling these three functions in the constructor of this class
+     * would be too early, resulting in non-paginated tables
+     */
+    public initTable(): void {
+        this.dataSource = new MatTableDataSource();
+        const me = this;
+        this.setFilter();
+        this.importer.getNewEntries().subscribe(newEntries => {
+            me.dataSource.data = newEntries;
+            if (newEntries.length){
```

Comment:
maybe a `me.hasFile = newEntries.length > 0;`.
Also check, if the `me` is necessary.